pkg/core/bundle_list.go

```diff
@@ -12,52 +13,128 @@ import (
 )
 
 const (
-	maxBundlesToList = 1000000
+	maxBundlesToList  = 1000000
+	typicalBundlesNum = 1000
 )
 
+type bundleChan struct {
```

Comment:
i'd typically been using two channels, one for error and one for results rather than multiplexing both errors and results through one channel.  the pattern of using one channel for both errors and results is okay with me, except could we call these structs `*ChanVal` or something?  i would expect the `*Chan` to be the channel itself, not the value it contains.

Reply:
dual channel is an interesting option. Giving it a try

Reply:
* [x] removed channels of struct (`bundleChan chan<-model.Descriptor`, èrrorChan chan<-error`, `keysChan <- chan string`)
* [x] used 2 channels instead of one as struct